### PR TITLE
Rdm 5691

### PR DESCRIPTION
**Pull request by smathangi**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDM-5691

### Change description ###

Update Existing cases with new [LAST_STATE_MODIFIED_DATE] metadata

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: lastStateModifiedMigration.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
```
Comment (smathangi) [inline lastStateModifiedMigration.sh]: Throw away endpoint after we run on production once.

File: src/main/java/uk/gov/hmcts/ccd/domain/service/createevent/CreateCaseEventService.java [hunk 2]
```diff
@@ -216,7 +226,7 @@ private void saveAuditEventForCaseDetails(final AboutToSubmitCallbackResponse ab
         auditEvent.setUserId(user.getId());
         auditEvent.setUserLastName(user.getSurname());
         auditEvent.setUserFirstName(user.getForename());
-        auditEvent.setCreatedDate(LocalDateTime.now(ZoneOffset.UTC));
+        auditEvent.setCreatedDate(now());
```
Comment (MSancaktutar) [inline src/main/java/uk/gov/hmcts/ccd/domain/service/createevent/CreateCaseEventService.java]: createdDate in the audit event may potentially be different from newCaseDetails.setLastStateModifiedDate(now).
Would this not be a problem?

File: src/main/java/uk/gov/hmcts/ccd/endpoint/LastStateModifiedMigrationEndpoint.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+    @Autowired
+    public LastStateModifiedMigrationEndpoint(LastStateModifiedMigrationService lastStateModifiedMigrationService) {
+        this.service = lastStateModifiedMigrationService;
+    }
+
+    @PostMapping(value = "/last-state-modified/migrate")
+    public void migrate(@RequestParam(value = "jurisdiction") final String jurisdiction,
+                                           @RequestParam(value = "batchSize", required = false, defaultValue = "5000") final int batchSize,
+                                           @RequestParam(value = "dryRun", required = false, defaultValue = "false") final boolean dryRun) {
```
Comment (MSancaktutar) [inline src/main/java/uk/gov/hmcts/ccd/endpoint/LastStateModifiedMigrationEndpoint.java]: dryRun default value should be true.

File: src/main/resources/db/changelog/db.changelog-RDM-4641.xml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
+
+
+    <changeSet id="rdm-4641" author="sateesh.mathangi@hmcts.net">
+        <addColumn tableName="case_data">
+            <column name="last_state_modified_date" type="timestamp">
+                <constraints nullable="true" />
```
Comment (MSancaktutar) [inline src/main/resources/db/changelog/db.changelog-RDM-4641.xml]: This should be changed to nullable="false", but maybe after the migration

Comment (smathangi) [reply src/main/resources/db/changelog/db.changelog-RDM-4641.xml]: Yes, after migration.

File: src/main/java/uk/gov/hmcts/ccd/endpoint/LastStateModifiedMigrationEndpoint.java (hunk 3 above)
Comment (smathangi) [reply src/main/java/uk/gov/hmcts/ccd/endpoint/LastStateModifiedMigrationEndpoint.java]: done

File: src/main/java/uk/gov/hmcts/ccd/domain/service/createevent/CreateCaseEventService.java (hunk 2 above)
Comment (smathangi) [reply src/main/java/uk/gov/hmcts/ccd/domain/service/createevent/CreateCaseEventService.java]: I thought about it. But I guess I don't think it'll give a problem. The difference is in just nano seconds. Similar thing we've already in `setLastModified` in  
https://github.com/hmcts/ccd-data-store-api/blob/master/src/main/java/uk/gov/hmcts/ccd/data/casedetails/DefaultCaseDetailsRepository.java#L70
Having said I'm happy to fix for `LastStateModified`. Please let me know.

Comment (smathangi) [reply src/main/java/uk/gov/hmcts/ccd/domain/service/createevent/CreateCaseEventService.java]: Done :)

Comment (MSancaktutar) [reply src/main/java/uk/gov/hmcts/ccd/domain/service/createevent/CreateCaseEventService.java]: As long as there is no and will be no logic in code or in data analysis/reporting that assumes strict equality of the values in both places, nano seconds wouldn't really matter. But my worry was around potential later issues with codes or data analytics assuming strict equality. Thanks for the improvement.

**Pull request closed without merging**